product_custom_attribute/models/attribute_attribute.py

```diff
@@ -47,8 +47,15 @@ def _build_attributes_notebook(self, vals):
 
     @api.multi
     def write(self, vals):
-        """ When deleting an attribute.option,delete all the attribute values
-        in existing products"""
+        """ When updating the domain field or deleting an attribute.option delete
+        all the related attribute values in existing products"""
+
+        if ("domain" in list(vals.keys()) and self.relation_model_id and
```

Comment:
vals.get('domain', False) instead of "domain" in list(vals.keys())

domain can't be None or "[]", but can it be [] or False ?

Reply:
You're right, domain can be '[]' or False, so as I want to delete the current options if a new domain "different from False or []" is registered, a better way to write my condition would be :
```
if vals.get('domain', False) != '[]' and self.relation_model_id:
```
That's right ?

Reply:
After looking closely I don't get it well what / why you are doing here.

About "
        """ When updating the domain field or deleting an attribute.option delete
        in existing products"""
        all the related attribute values in existing products""" 

Why ? If domain changes, you may not want to impact all products tied to the values. attributes can be the same with a different domain.

Then, why did you choose to not you use a @api.depends('domain', 'relation_model_id') ?
The removal of the option_ids can be done in unlink I guess.

Reply:
As we talked yesterday, [this commit](https://github.com/shopinvader/odoo-pim/commit/dd77b5b53feb6ffa1df54cb61b7e8d1c356722a7) might resolve the question and delete the existing option values in product ONLY if this value is not in the new domain/options available.